staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -569,6 +592,54 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	return nil
 }
 
+// IsStarted reports whether the informer has already been started
+func (s *sharedIndexInformer) IsStarted() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.started
+}
+
+// IsStopped reports whether the informer has already been stopped
+func (s *sharedIndexInformer) IsStopped() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.stopped
+}
+
+// EventHandlerCount reports whether the informer still has registered
+// event handlers
+func (s *sharedIndexInformer) EventHandlerCount() int {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return len(s.processor.listeners)
+}
+
+// RemoveEventHandler tries to remove a formerly added event handler.
+// Only event handlers that are (go) comparable can be removed.
+// Please note: If, for some reason, the same handler has been added multiple
+// times, all registrations will be removed.
+func (s *sharedIndexInformer) RemoveEventHandler(handler ResourceEventHandler) error {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+
+	if s.stopped {
+		return fmt.Errorf("Handler %v is not removed from shared informer because it has stopped already", handler)
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

error string is better to start with lowercase